pkg/jsontabwriter/jsontabwriter.go

```diff
@@ -0,0 +1,161 @@
+package jsontabwriter
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/ionos-cloud/ionosctl/v6/pkg/constants"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/json2table"
+	"github.com/spf13/viper"
+)
+
+var outputFormatErr = "invalid format: %s"
+
+// TODO: remove cols as function parameter once --cols flag fix is ready
+func GenerateOutput(rootPath string, jsonPaths map[string]string, obj interface{}, cols []string) (string, error) {
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		return generateJSONOutput(obj)
+	}
+
+	if outputFormat == "text" {
+		return generateTextOutput(rootPath, obj, jsonPaths, cols, false)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
+
+func GenerateOutputPreconverted(obj interface{}, convertedObj []map[string]interface{}, cols []string) (string, error) {
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		return generateJSONOutput(obj)
+	}
+
+	if outputFormat == "text" {
+		return generateTextOutput("", convertedObj, nil, cols, true)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
+
+func generateJSONOutput(obj interface{}) (string, error) {
+	out, err := json.MarshalIndent(obj, "", "\t")
+	if err != nil {
+		return "", err
+	}
+
+	return string(out) + "\n", nil
+}
+
+func generateTextOutput(rootPath string, obj interface{}, jsonPaths map[string]string, cols []string, skipConversion bool) (string, error) {
+	var table []map[string]interface{}
+	var err error
+
+	if skipConversion {
+		table = obj.([]map[string]interface{})
+	} else {
+		table, err = json2table.ConvertJSONToTable(rootPath, jsonPaths, obj)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	var buff = new(bytes.Buffer)
+	var w = new(tabwriter.Writer)
+	w.Init(buff, 5, 0, 3, ' ', tabwriter.StripEscape)
+
+	if !viper.IsSet(constants.ArgNoHeaders) {
+		if _, err = fmt.Fprintln(w, strings.Join(cols, "\t")); err != nil {
+			return "", nil
```

Comment:
This should probably return an error

```suggestion
			return "", fmt.Errorf("failed writing columns: %w", err)
```